hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/DefaultDataSourceProvider.java

```diff
@@ -43,14 +52,26 @@
    */
   @Override
   public DataSource get() {
```

Comment:
This is actually an antipattern :-) Although I am guilty of this as well. Instead of the if were should have a DerbyDataSourceProvider, SqliteDataSourceProvider and Default one.

Reply:
Yeah, I agree. Will look into this change. Maybe I can create 2 different datasource providers and re-use in the test.